component/prometheus/integrations/node_exporter/node_exporter.go

```diff
@@ -0,0 +1,101 @@
+package node_exporter
+
+import (
+	"context"
+	"net/http"
+	"path"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	node_integration "github.com/grafana/agent/pkg/integrations/node_exporter"
+	"github.com/prometheus/common/model"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:    "prometheus.integration.node_exporter",
+		Args:    node_integration.Config{},
+		Exports: Exports{},
```

Comment:
This is our first component which must be a singleton until #1861 is resolved. We need to set `Singleton: true` here to mark it as one.